src/pages/TasksDemo.tsx

```diff
@@ -0,0 +1,102 @@
+
+import { Box, Button, Flex, Heading, Input, Stack, HStack, Spinner } from '@chakra-ui/react';
+import { collection, addDoc, updateDoc, deleteDoc, doc, query, CollectionReference, Timestamp, DocumentReference } from 'firebase/firestore';
+import { useState } from 'react';
+import { useCollection } from 'react-firebase-hooks/firestore';
+import { Task } from '../models/Task';
+
+import { db } from '../../firebase.config';
+
+export function TasksPageDemo() {
+  const [newTask, setNewName] = useState('');
+  const [newDeadline, setNewDeadline] = useState<Timestamp | null>(null);
+  const [newDescription, setNewDescription] = useState('');
+  const [newImportance, setNewImportance] = useState(0);
+
+  const tasksCollectionRef = collection(db, 'tasks');
+
+  const [tasks, tasksLoading, tasksError] = useCollection(query(tasksCollectionRef) as CollectionReference<Task>);
+
+  const createTask = async () => {
+    await addDoc(tasksCollectionRef, { name: String(newTask), deadline: newDeadline, description: newDescription, activated: true, importance: Number(newImportance), status: true, uuid: 'test' });
+  };
+  const updateTask = async (id: string, updatedName: string, updatedDescription: string, updatedDeadline: Timestamp) => {
+    const taskDoc = doc(db, 'tasks', id);
+    const newFields = { name: updatedName, description: updatedDescription, deadline: updatedDeadline };
+    await updateDoc(taskDoc, newFields);
+  };
+  const deleteTask = async (id: string) => {
+    const taskDoc = doc(db, 'tasks', id);
+    await deleteDoc(taskDoc);
+  };
```

Comment:
It is also possible to move CRUD (or probably CUD) functionality to a custom hook, which would be a good exercise to work with React hooks though it is not required and good enough as it is.